### PR TITLE
fix(person-modal): fix initial person modal request property filters

### LINKED ISSUE #10964: Unique Persons not reporting correctly
## Bug description

When looking at the Insights report, I can see that there are unique pageviews. And I can also see in the Persons & Groups that there are unique users. But when I click to view people in the Insights report it says that there are 0 unique persons.

## How to reproduce

I've recorded the issue here:
https://app.userback.io/viewer/1/70836/1920864N8EYWSEH/

## Environment

[x] PostHog Cloud

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

https://posthog.slack.com/archives/C02KGGDCA6Q/p1659969290272539

Fixes https://github.com/PostHog/posthog/issues/10964

## Changes

Fixes a bug with today's actor modal where request properties were malformed.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Manual QA

### CODE REVIEW

File: frontend/src/scenes/trends/personsModalLogic.ts [hunk 1]
```diff
@@ -84,8 +84,8 @@ export function parsePeopleParams(peopleParams: PeopleParamType, filters: Partia
             { key: params.breakdown, value: breakdown_value, type: 'event' } as PropertyFilter,
         ]
     }
-    if (action?.properties) {
-        params.properties = { ...(flattenedPropertyGroup || {}), ...action.properties }
+    if (action?.properties && (action.properties?.length ?? 0) > 0) {
```
Comment (neilkakkar) [inline frontend/src/scenes/trends/personsModalLogic.ts:87]: Hmm, not sure if this doesn't introduce other new bugs. Getting very confusing..

See in the line above, where if we're dealing with a breakdown, we set `params.properties` to include the breakdown filter value, which is necessary for showing events only for the selected breakdown value.

And here, we're overriding that setting with properties on the action? (not sure if these are global filters or entity filters).

In any case, this feels a little bit better, as atleast `params.properties` are consistent now.

However, another case worth keeping in mind is that if `params.properties` originally were an OR property group, the flattening turns it into an AND, which will again return wrong results.

**Review by neilkakkar — APPROVED** [review]
Approving, but it seems like there's still quite a few bugs around this code.

Having unit tests around this logic would massively ~~simplify~~ de-confuse things & prevent regressions, as I'm pretty sure this isn't the first time we're tweaking this specific piece of code to fix a bug (while creating other bugs, for example, for when properties here is supposed to be a dict 🤷 )

**Comment by alexkim205** [comment]
Yup agreed I think personsModalLogic and its corresponding component needs a major refactor with proper testing to account for these cases. For the purposes of this fix, I'd prefer to get this in quickly and then cut a separate ticket for the refactor since a lot of users have reported this specific bug

File: frontend/src/scenes/trends/personsModalLogic.ts (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/trends/personsModalLogic.ts:87]: Hm, I actually don't think this comment applies any more 

```
    // If breakdown type is cohort, we use breakdown_value
    // If breakdown type is event, we just set another filter
```

We might be able to get away with getting rid of the whole flattening nonsense

**Comment by alexkim205** [comment]
Update removed flattening transformation entirely so that we don't run into any weird compatibility bugs anymore since our API support property groups now.

**Pull request merged**